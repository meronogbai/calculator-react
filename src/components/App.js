import '../App.css';
import { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operatiion: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    this.setState(state => calculate(
      { total: state.total, next: state.next, operatiion: state.operatiion },
      buttonName,
    ));
  }

  render() {
    const { total, next } = this.state;
    return (
      <div className="App">
        <Display result={next || total || '0'} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}
