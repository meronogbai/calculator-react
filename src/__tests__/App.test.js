import { create } from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../components/App';

test('renders correctly', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays numbers', () => {
  render(<App />);
  const button = screen.getByText('9');
  expect(button).toBeInTheDocument();
});

test('displays operators', () => {
  render(<App />);
  const button = screen.getByText(/\+$/i);
  expect(button).toBeInTheDocument();
});

test('displays result of calculation', () => {
  render(<App />);
  fireEvent.click(screen.getByText('7'));
  fireEvent.click(screen.getByText('X'));
  fireEvent.click(screen.getByText('6'));
  fireEvent.click(screen.getByText('='));
  const result = screen.getByText('42');
  expect(result).toBeInTheDocument();
});

test('clears display on clicking AC', async () => {
  render(<App />);
  fireEvent.click(screen.getByText('7'));
  fireEvent.click(screen.getByText('X'));
  fireEvent.click(screen.getByText('6'));
  fireEvent.click(screen.getByText('='));
  fireEvent.click(screen.getByText('AC'));
  const zeros = await screen.findAllByText('0');
  expect(zeros).toHaveLength(2);
});
