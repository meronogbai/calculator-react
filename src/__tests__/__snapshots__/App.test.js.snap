// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="Display"
  >
    0
  </div>
  <div
    className="ButtonPanel"
  >
    <div
      className="group-1"
    >
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="Button operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="group-2"
    >
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="Button operator"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="group-3"
    >
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="Button operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="group-4"
    >
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="Button operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="group-5"
    >
      <button
        className="Button zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="Button operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
